RecoTracker/LSTCore/src/alpaka/Event.dev.cc

```diff
@@ -1333,28 +1348,24 @@ ObjectRangesBuffer<alpaka_common::DevHost>& Event::getRanges(bool sync) {
   return rangesInCPU_.value();
 }
 
-MiniDoubletsBuffer<alpaka_common::DevHost>& Event::getMiniDoublets(bool sync) {
-  if (!mdsInCPU_) {
-    // Get nMemoryLocations parameter to initialize host based mdsInCPU_
-    auto nMemHost_buf_h = cms::alpakatools::make_host_buffer<unsigned int[]>(queue_, 1u);
-    alpaka::memcpy(queue_, nMemHost_buf_h, miniDoubletsBuffers_->nMemoryLocations_buf);
-    alpaka::wait(queue_);  // wait for the value before using
-
-    auto const nMemHost = *nMemHost_buf_h.data();
-    mdsInCPU_.emplace(nMemHost, nLowerModules_, cms::alpakatools::host(), queue_);
-    mdsInCPU_->setData(*mdsInCPU_);
-
-    alpaka::memcpy(queue_, mdsInCPU_->nMemoryLocations_buf, miniDoubletsBuffers_->nMemoryLocations_buf);
-    alpaka::memcpy(queue_, mdsInCPU_->anchorHitIndices_buf, miniDoubletsBuffers_->anchorHitIndices_buf, nMemHost);
-    alpaka::memcpy(queue_, mdsInCPU_->outerHitIndices_buf, miniDoubletsBuffers_->outerHitIndices_buf, nMemHost);
-    alpaka::memcpy(queue_, mdsInCPU_->dphichanges_buf, miniDoubletsBuffers_->dphichanges_buf, nMemHost);
-    alpaka::memcpy(queue_, mdsInCPU_->nMDs_buf, miniDoubletsBuffers_->nMDs_buf);
-    alpaka::memcpy(queue_, mdsInCPU_->totOccupancyMDs_buf, miniDoubletsBuffers_->totOccupancyMDs_buf);
-    if (sync)
-      alpaka::wait(queue_);  // host consumers expect filled data
+template <typename TSoA, typename TDev>
+typename TSoA::ConstView Event::getMiniDoublets(bool sync) {
+  if constexpr (std::is_same_v<TDev, DevHost>) {
+    return miniDoubletsDC_->const_view<TSoA>();
```

Comment:
can this be replaced by `std::is_same_v<Platform, alpaka::PlatformCpu>` to use what comes with the alpaka namespace already. With this I think the template is not needed here for the `getMiniDoublets` method

Reply:
I'll try it out, but I don't think we can avoid the template

Reply:
as I mentioned during the meeting, I agree that this effectively requires SFINAE. So, the current implementation is OK here.